### PR TITLE
Add support for .NET 8.0

### PR DESCRIPTION
**Pull request by scott-xu**

### CODE REVIEW

**Comment by scott-xu** [comment]
Was going to add source link to SSH.NET and just found that it is [included in the .NET 8 SDK and enabled by default]( https://github.com/dotnet/sourcelink#using-source-link-in-net-projects).
This PR is only to add support for .NET 8.0. I can create another PR afterwards, to generate the [symbol package](https://docs.microsoft.com/en-us/nuget/create-packages/symbol-packages-snupkg).
Would you please kindly have a review? Thanks! @WojciechNagorski @drieseng

File: appveyor.yml [hunk 1]
```diff
@@ -12,20 +12,20 @@ for:
       - image: Ubuntu2204
 
   install:
-    - sh: sudo apt-get update && sudo apt-get install -y dotnet-sdk-7.0=7.0.403-1
+    - sh: sudo apt-get update && sudo apt-get install -y dotnet-sdk-8.0
```
Comment (WojciechNagorski) [inline appveyor.yml]: I would like to use the explicit version `dotnet-sdk-8.0=8.0.100` (or `8.0.100-1` you have to check it). When Microsoft releases a new version our tests will no longer work.

**Review by WojciechNagorski — CHANGES_REQUESTED** [review]

File: appveyor.yml (hunk 1 above)
Comment (scott-xu) [reply appveyor.yml]: From reading the console output, I know it is `8.0.100-1`.

Comment (Rob-Hague) [reply appveyor.yml]: The tests will work fine with a later version. Having an explicit version is just extra maintenance

File: test/Renci.SshNet.Tests/Classes/BaseClientTest_Connected_KeepAliveInterval_NotNegativeOne.cs [hunk 2]
```diff
@@ -56,12 +56,12 @@ protected override void Act()
...
             _client.KeepAliveInterval = _keepAliveInterval;
 
-            // allow keep-alive to be sent a few times. .NET 7 is faster and
+            // allow keep-alive to be sent a few times. .NET 8 is faster and
             // we need to wait less because we want exactly three messages in a session.
 #if NETFRAMEWORK
             Thread.Sleep(195);
 #else
-            Thread.Sleep(180);
+            Thread.Sleep(170);
```
Comment (Rob-Hague) [inline test/Renci.SshNet.Tests/Classes/BaseClientTest_Connected_KeepAliveInterval_NotNegativeOne.cs:64]: We should really get rid of timing in tests...

Comment (scott-xu) [reply test/Renci.SshNet.Tests/Classes/BaseClientTest_Connected_KeepAliveInterval_NotNegativeOne.cs:64]: Agree. Better to create another PR...

File: appveyor.yml (hunk 1 above)
Comment (scott-xu) [reply appveyor.yml]: If the unit test won't work for newer SDK, it means either our unit test can improve or it is a real problem found.

**Review by WojciechNagorski — APPROVED** [review]
LGTM! 
@drieseng @Rob-Hague do you mind?

**Comment by Rob-Hague** [comment]
It's fine for me. I'm not convinced we need a net8.0 target without using any .NET 8 features (other than throw helpers), but using the .NET 8 SDK for source-link is good 👍

**Comment by WojciechNagorski** [comment]
The 2023.0.1 version has been released to Nuget: https://www.nuget.org/packages/SSH.NET/2023.0.1

**Pull request merged**